float32 angleX
float32 distance
